core/types/l1_message_tx.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+)
+
+const L1MessageTxType = 0x7E
+
+// payload, RLP encoded
+type L1MessageTx struct {
+	Nonce  uint64
+	Gas    uint64          // gas limit
+	To     *common.Address `rlp:"nil"` // nil means contract creation
```

Comment:
What does `rlp:"nil"` do?

Reply:
indicates to the RLP package that when the To field is nil, the RLP encoder should treat the field as an empty byte array. added because the other transaction types have it as well

Reply:
We don't allow enforced contract creation from L1?

Reply:
Why we don't allow contract creation from L1? @zimpha

Reply:
will reopen on next PR